src/main/java/com/redhat/cloud/notifications/utils/TimeAgoFormatter.java

```diff
@@ -0,0 +1,73 @@
+package com.redhat.cloud.notifications.utils;
+
+import java.time.LocalDateTime;
+import java.time.Period;
+import java.time.temporal.ChronoField;
+import java.time.temporal.TemporalField;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TimeAgoFormatter {
+
+    private static class Configuration {
+        private final TemporalField temporalField;
+        private final String singular;
+        private final String pluralFormat;
+
+        Configuration(TemporalField temporalField, String singular, String pluralFormat) {
+            this.temporalField = temporalField;
+            this.singular = singular;
+            this.pluralFormat = pluralFormat;
+        }
+
+    }
+
+    private static final List<Configuration> config = List.of(
+            new Configuration(ChronoField.YEAR, "1 year ago", "%d years ago"),
+            new Configuration(ChronoField.MONTH_OF_YEAR, "1 month ago", "%d months ago"),
+            new Configuration(ChronoField.DAY_OF_MONTH, "1 day ago", "%d days ago"),
```

Comment:
I don't think `ChronoField.DAY_OF_MONTH` will work if today is `2021-01-01` and the date to format is `2020-31-12`. The `naturalUnit` variable should then be equal to `1 - 31`, which will lead to `Today` being returned while it should be `1 day ago`. The same kind of issue should happen with `ChronoField.MONTH_OF_YEAR`.

Reply:
Thanks, I'll add tests for that.

Reply:
So this is what insights do: 
https://github.com/RedHatInsights/frontend-components/blob/86b02e3c206663d640d30b32087061c04db1595d/packages/components/src/DateFormat/helper.js#L12-L19

I'll adjust to match that behavior.